stages/org.osbuild.rhsm.facts

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/python3
+"""
+This stage allows storing artifact-properties of the Red Hat Subscription
+Manager (rhsm-facts) in the built image.
+"""
+
+import json
+import os
+import sys
+
+import osbuild.api
+
+
+SCHEMA = r"""
+"properties": {
+  "facts": {
+    "type": "object",
+    "description": "Map of Red Hat Subscription Manager facts.",
+    "additionalProperties": true
+  }
+}
+"""
+
+
+def main(tree, options):
+    path = os.path.join(tree, "usr/share/osbuild/self")
+    file = os.path.join(path, "rhsm.facts")
+
+    os.makedirs(path, exist_ok=True)
+
+    with open(file, "x", encoding="utf-8") as f:
+        json.dump(options["facts"], f)
```

Comment:
Is the format really arbitrary JSON? If so: 👍 , but if there are restrictions, we should enforce them.

Reply:
"Any user-defined JSON files that override or add system facts to determine entitlement compatibility. Any facts files must end in .facts."

That's all I got.

Reply:
Should we reinforce the key prefix somehow, or have that as a separate option? It's not a strict requirement but it's definitely useful to have all our facts share the same prefix, also to avoid any conflicts between other custom facts. Then again we can also just make sure that composer always uses the prefix.

Reply:
If the tool places no restriction I think we should keep the policy in composer. @gicmo?

Reply:
We could make it easy to use from composer by having a `prefix` option that would get applied to all the keys. But I guess we can also just use the go json annotation a la `Version string 'json:"image_builder.version"'`